types/yup/yup-tests.ts

```diff
@@ -459,6 +468,21 @@ const testObject: MyInterface = {
 
 typedSchema.validateSync(testObject); // $ExpectType MyInterface
 
+// shape function
+interface InterfaceA {
+    aField: string;
+}
+
+interface InterfaceB {
+    bField: number;
+}
+
+const definitionA: yup.ObjectSchemaDefinition<InterfaceA> = { aField: yup.string() };
+const definitionB: yup.ObjectSchemaDefinition<InterfaceB> = { bField: yup.number() };
+let combinedSchema = yup.object(definitionA).shape(definitionB); // $ExpectType ObjectSchema<InterfaceA & InterfaceB>
+
+combinedSchema = yup.object<InterfaceA>({ aField: yup.string() }).shape<InterfaceB>({ bField: yup.number() });
```

Comment:
I would like to see a test of the overlapping fields with different types as well

Reply:
Agreed.

Reply:
Such a test is now included in the PR.